interview-questions.md

```diff
@@ -7,53 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
+  Researched answer: Ruby blocks is an anonumous functions that can be passed into methods. It consists of chunks of codes and a name is assign to a block.
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: I think that gem is a library for ruby on rails where data is stored to be pulled out anytime the user wants someting from it. It is encrypted inside ruby on rails and is automatically installed ones rails has been installed. I don't programmer needed to access gem and the rails automatically use it when needed.
 
-  Researched answer:
+  Researched answer: Gems are libraries that allow any ruby on rails developer to add functionalities without the need to write code. It is also a plugins for adding features.
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: It is a web application framework that is use to connect frontend to database to backend. This is a tool create to support ruby and be use whenever ruby is used for web development. By installing ruby on rails, it is easier to use database and ruby together when storing data into database.
 
-  Researched answer:
+  Researched answer: Ruby on rails is a server side web application framework use for ruby programming language. It is a full stack framework that contains all the tools needed to build amazing web appps on both the font and the backend.
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: Relational database is a database of information consisting of multiple table inside it. Each table consist of rows and columns where columns pertains to a name, age, or information of something and rows is the instance of object of that information. This is also where all the information is stored after the user input their information and any other data needed for the web page.
 
-  Researched answer:
+  Researched answer: It is like and excel documents where information is stored in rows and columns. Instead of one big table, this consists of multiple table that have relationships with each other.
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are use in database and is a unique key for every data. This is important because it determines the difference between data. If two data has the same exact information, without the primary key attach to it, the data would then be considered duplicate. But with this primary key, two data with the same information can be determine base on thier unique keys attach to them.
 
-  Researched answer:
+  Researched answer: It is a database table column that identifies the uniqueness of each table record. It is used as a unique identifier to quickly parse data within the table.
```

Comment:
Awesome answers in this section!  Really detailed and thorough for both your answers and your researched answers.  Well done!